### ISSUE TITLE
SSA and generics (go1.18)

### ISSUE THREAD

**Issue by ldez** [ryanrolds/sqlclosecheck#18]
Currently, SSA is not working with generics.

So your linter produces a panic when it is used with generics.

There is an issue open about that in the Go repository: https://github.com/golang/go/issues/48525

Inside golangci-lint, we have disabled your linters: https://github.com/golangci/golangci-lint/issues/2649

You have 2 solutions:
- waiting for a version of SSA that will support generics
- dropping the SSA analyzers and using something else to analyze the code.

Related to https://github.com/golang/go/issues/50558

**Comment by ldez** [comment]
SSA has been updated and it seems to work inside golangci-lint.

**Comment by ldez** [comment]
The code of the linter needs to be adapted to manage generics:

```go

func SqlCloseCheck(db *sql.DB, a int) {
	rows, _ := db.Query("select id from tb") // <- detected OK
	for rows.Next() {

	}
}

func SqlCloseCheckG[T ~int64](db *sql.DB, a T) {
	rows, _ := db.Query("select id from tb") // <- not detected KO
	for rows.Next() {

	}
}
```

**Comment by ldez** [comment]
go1.19 will come in about 2 weeks, it can be good if you can fix your linter before this release.

**Comment by dolmen** [comment]
@ldez For what I see above, it seems that `sqlclosecheck` works on code that doesn't use generics. Am I right?

So would it be possible to enable `sqlclosecheck` in `golangci-lint` to have that feature back, at least to check non-generic code?

**Comment by ldez** [comment]
you can add the following option in your configuration:
```
run:
  # ...
  go: go1.17
```

**Comment by ryanrolds** [comment]
I believe I've resolved the panic and it's detecting the [provided Generics test case](https://github.com/ryanrolds/sqlclosecheck/issues/18#issuecomment-1189618981). A pre-release has been cut and I will be testing on some personal projects to build confidence. 

Thank you for your patience. I will close after building some confidence.
